docs/provider/index.rst

```diff
@@ -104,3 +104,4 @@ Though it is a normal object, not all attributes are analyzed by ``Container``,
    alias
    from_context
    decorate
+   join_providers
```

Comment:
I am not sure that it should be in same list as mentioned functions, or even a separate file. Let's move somewhere (right here,  to provide_all, or just mention in other places)